### PR TITLE
Feat(#314):-Update feedback schema

### PR DESCRIPTION
**Pull request by anjneeksharma**
# Closes/Fixes/Resolves

### 

# Explain the feature/fix

Updated feedback schema and removed timestamp.
## Does this PR introduce a breaking change

No

## Screenshots

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

**Comment by anjneeksharma** [comment]
@deshmukhmayur  its just a minimal PR i updated and request you to comment your suggestion.

File: packages/feedback-service/src/schema.ts [hunk 1]
```diff
@@ -5,18 +5,10 @@ const FeedbackSchema: Schema = new Schema({
...
-      name: String,
-    },
-    modifiedAt: Date,
-    modifiedBy: {
-      kerberosID: String,
-      name: String,
-    }
-  },
+  createdAt: { type: Date, default: Date.now },
+  createdBy: { type: String, ref: 'User' },
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/schema.ts:9]: This should be just a string, not a ref. Because you do not have access to the User db. So you need to store the user's rhuuid here, and in your resolver, you'll have to fetch the user details from the User microservice (using it's APIs).

_Also a note: Each resolver should send just 1 query to get whatever data you require._

File: packages/feedback-service/src/schema.ts [hunk 2]
```diff
@@ -5,18 +5,10 @@ const FeedbackSchema: Schema = new Schema({
...
-      kerberosID: String,
-      name: String,
-    },
-    modifiedAt: Date,
-    modifiedBy: {
-      kerberosID: String,
-      name: String,
-    }
-  },
+  createdAt: { type: Date, default: Date.now },
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/schema.ts:8]: Also, change this field name to `createdOn`

File: packages/feedback-service/src/schema.ts [hunk 3]
```diff
@@ -5,18 +5,10 @@ const FeedbackSchema: Schema = new Schema({
...
+  createdBy: { type: String, ref: 'User' },
+  modifiedAt: Date,
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/schema.ts:10]: And change this to `updatedOn`
```suggestion
  updatedOn: { type: Date, default: Date.now },
```

File: packages/feedback-service/src/schema.ts [hunk 4]
```diff
@@ -5,10 +5,10 @@ const FeedbackSchema: Schema = new Schema({
...
   ticketID: String,
   spa: String,
-  createdAt: { type: Date, default: Date.now },
-  createdBy: { type: String, ref: 'User' },
-  modifiedAt: Date,
-  modifiedBy: { type: String, ref: 'User' },
+  createdOn: { type: Date, default: Date.now },
+  createdBy: String,
+  updatedOn: { type: Date, default: Date.now },
+  modifiedBy: String,
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/schema.ts:11]: As discussed yesterday, replace all `modifiedAt` and `modifiedBy` with `updatedOn` and `updatedBy`

```suggestion
  updatedBy: String,
```

File: packages/feedback-service/src/resolver.ts [hunk 5]
```diff
@@ -7,12 +7,14 @@ export const FeedbackResolver = {
   Query: {
     listFeedback(root: any, args: any, ctx: any) {
       return Feedback.find()
+        .populate('createdBy')
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:10]: This will no longer work. Did you test the APIs before committing?

Also, here you need to fetch the user details from User/Group Microservice using its API.
And add it to the feedback response and return it.

And the part about sending a single request, all the user data that is needed for the feedback response should be fetched using a single GraphQL query.
Check `getUsersBy` query in User/Group Microservice. (#281)

File: packages/feedback-service/src/resolver.ts [hunk 6]
```diff
@@ -1,17 +1,50 @@
...
+          const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: graphql
+          };
+          fetch('http://localhost:8080/graphql', requestOptions)
+          .then((result: any) => result.json())
+          .then((output: any) => {
+            const usersdata = output.data;
+            for (const res of response) {
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:32]: Use array.map instead of the for loop here.

File: packages/feedback-service/src/resolver.ts [hunk 7]
```diff
@@ -1,17 +1,50 @@
...
+              if (usersdata['rhatUUID_' + res['createdBy']]) {
+                const createdby = usersdata['rhatUUID_' + res['createdBy']];
+                const updatedby = usersdata['rhatUUID_' + res['updatedBy']];
+                res['createdBy'] = JSON.stringify(createdby);
+                res['updatedBy'] = JSON.stringify(updatedby);
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:37]: The entire object should be passed here.
```suggestion
                res['createdBy'] = createdby;
                res['updatedBy'] = updatedby;
```

File: packages/feedback-service/src/typedef.graphql [hunk 8]
```diff
@@ -1,5 +1,37 @@
 scalar DateTime
 
+type createdBy {
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:3]: Just create a single type called UserType and use it for both `createdBy` and `updatedBy`

```suggestion
type UserType {
```

File: packages/feedback-service/src/typedef.graphql [hunk 9]
```diff
@@ -44,7 +76,7 @@ type TicketType {
 
 type Query {
   # Fetches all Feedbacks
-  listFeedback: [FeedbackType]
+  listFeedback: [FeedbackWithUserType]
   # Fetches specific Feedback by its id
   getFeedback(id: String!): FeedbackType
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/typedef.graphql:81]: You also need to do the same for getFeedback and any other queries that are returning feedback.

File: packages/feedback-service/src/resolver.ts [hunk 10]
```diff
@@ -1,17 +1,50 @@
...
+          variables: {}
+          });
+          const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: graphql
+          };
+          fetch('http://localhost:8080/graphql', requestOptions)
+          .then((result: any) => result.json())
+          .then((output: any) => {
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:30]: Do not use `any`. Make use of the proper types.

```suggestion
          .then(result => result.json())
          .then(output => {
```

File: packages/feedback-service/src/resolver.ts (hunk 7 above)
Comment (anjneeksharma) [reply packages/feedback-service/src/resolver.ts]: i tried rebuilding it and then run but still i am not able to assign it.
                res['createdBy'] = createdby;

File: packages/feedback-service/src/resolver.ts (hunk 10 above)
Comment (anjneeksharma) [reply packages/feedback-service/src/resolver.ts]: This is API response even i tried creating its type its not working.

File: packages/feedback-service/.env.example [hunk 11]
```diff
@@ -25,3 +25,6 @@ COOKIE_PARSER=
 
 ## Authorization Token
 # AUTH_TOKEN=
+
+USER_SERVICE_SERVICE_HOST=172.30.246.47
```
Comment (deshmukhmayur) [inline packages/feedback-service/.env.example:29]:
```suggestion
USER_SERVICE_SERVICE_HOST=
```

**Comment by anjneeksharma** [comment]
@deshmukhmayur  plz have a another look.

File: packages/feedback-service/src/resolver.ts [hunk 12]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
               const usersdata = output.data;
-              newresponse = response.map(function (res) {
-                const createdby = usersdata['rhatUUID_' + res['createdBy']];
-                const updatedby = usersdata['rhatUUID_' + res['updatedBy']];
-                res['createdBy'] = createdby;
-                res['updatedBy'] = updatedby;
+              const newresponse = response.map(function (data: any) {
+                const createdby = usersdata['rhatUUID_' + data['createdBy']][0];
+                const updatedby = usersdata['rhatUUID_' + data['updatedBy']];
+                const res = { ...{ ...data }._doc, createdBy: createdby, updatedBy: updatedby };
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:34]: Can return the object directly. No need to store it in a variable if there're no operations to perform.

```suggestion
                return { ...{ ...data }._doc, createdBy: createdby, updatedBy: updatedby };
```

File: packages/feedback-service/src/resolver.ts [hunk 13]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
+            body: graphql
+          };
+          return fetch('http://localhost:8080/graphql', requestOptions)
+            .then((result: any) => result.json())
+            .then((output: any) => {
+              const usersdata = output.data;
+              const createdby = usersdata['rhatUUID_' + response['createdBy']][0];
+              const updatedby = usersdata['rhatUUID_' + response['updatedBy']];
+              const res = { ...{ ...response }._doc, createdBy: createdby, updatedBy: updatedby };
+              return res;
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:68]:
```suggestion
              return { ...{ ...response }._doc, createdBy: createdby, updatedBy: updatedby };
```

File: packages/feedback-service/src/resolver.ts [hunk 14]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
+          return fetch('http://localhost:8080/graphql', requestOptions)
             .then((result: any) => result.json())
             .then((output: any) => {
               const usersdata = output.data;
-              newresponse = response.map(function (res) {
-                const createdby = usersdata['rhatUUID_' + res['createdBy']];
-                const updatedby = usersdata['rhatUUID_' + res['updatedBy']];
-                res['createdBy'] = createdby;
-                res['updatedBy'] = updatedby;
+              const newresponse = response.map(function (data: any) {
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:31]:
```suggestion
              return response.map(function (data: any) {
```

File: packages/feedback-service/src/resolver.ts [hunk 15]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
             headers: { 'Content-Type': 'application/json' },
             body: graphql
           };
-          fetch('http://localhost:8080/graphql', requestOptions)
+          return fetch('http://localhost:8080/graphql', requestOptions)
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:27]: Use environment varables for the User Service here.
Do not commit development/debugging code to git.

File: packages/feedback-service/src/resolver.ts [hunk 16]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
-                res['updatedBy'] = updatedby;
+              const newresponse = response.map(function (data: any) {
+                const createdby = usersdata['rhatUUID_' + data['createdBy']][0];
+                const updatedby = usersdata['rhatUUID_' + data['updatedBy']];
+                const res = { ...{ ...data }._doc, createdBy: createdby, updatedBy: updatedby };
                 return res;
               });
+              return newresponse;
             })
             .catch((error: any) => console.log('error', error));
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:39]:
```suggestion
            .catch( ( error: any ) => { throw error; } );
```

File: packages/feedback-service/src/resolver.ts [hunk 17]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
+          return fetch('http://localhost:8080/graphql', requestOptions)
+            .then((result: any) => result.json())
+            .then((output: any) => {
+              const usersdata = output.data;
+              const createdby = usersdata['rhatUUID_' + response['createdBy']][0];
+              const updatedby = usersdata['rhatUUID_' + response['updatedBy']];
+              const res = { ...{ ...response }._doc, createdBy: createdby, updatedBy: updatedby };
+              return res;
+            })
+            .catch((error: any) => console.log('error', error));
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:70]:
```suggestion
            .catch( ( error: any ) => { throw error; } );
```

File: packages/feedback-service/src/resolver.ts [hunk 18]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
+          }
+        })
         .catch(err => err);
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:73]:
```suggestion
        .catch( ( error: any ) => { throw error; } );
```

File: packages/feedback-service/src/resolver.ts [hunk 19]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
+            const usersdata = output.data;
+            const newresponse = response.map(function (data: any) {
+              const createdby = usersdata['rhatUUID_' + data['createdBy']][0];
+              const updatedby = usersdata['rhatUUID_' + data['updatedBy']];
+              const res = { ...{ ...data }._doc, createdBy: createdby, updatedBy: updatedby };
+              return res;
+            });
+            return newresponse;
+          })
+          .catch((error: any) => console.log('error', error));
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:104]:
```suggestion
          .catch( ( error: any ) => { throw error; } );
```

File: packages/feedback-service/src/resolver.ts [hunk 20]
```diff
@@ -25,32 +24,86 @@ export const FeedbackResolver = {
...
+      })
+      .catch((error: any) => console.log('error', error));
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:106]:
```suggestion
      .catch( ( error: any ) => { throw error; } );
```

File: packages/feedback-service/src/resolver.ts [hunk 21]
```diff
@@ -11,7 +11,6 @@ export const FeedbackResolver = {
       return Feedback.find()
         .sort({ 'timestamp': -1 })
         .then(response => {
-          let newresponse: any[] = [];
           const createdBy = response.map(res => `rhatUUID_${res.createdBy}:getUsersBy(rhatUUID:"${res.createdBy}") { _id name title uid rhatUUID isActive}`);
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:14]: This might give an error for duplicate aliases.
Have you tested it for cases where the same user has submitted multiple feedbacks?

Check if it throws any error for multiple feedbacks with same `createdBy` or `updatedBy`.

Comment (deshmukhmayur) [reply packages/feedback-service/src/resolver.ts:14]: If it does throw an error, you may have to get the unique rhatUUIDs first, and then pass them on to the GraphQL request.

Comment (anjneeksharma) [reply packages/feedback-service/src/resolver.ts:14]: Yes i tested it and tested it again now duplicated aliases. does not through error.

**Comment by sayak-sarkar** [comment]
@anjneeksharma The commit history within this PR looks quite polluted with multiple rebase commits. Can you please clean up the commit history for this PR? @riginoommen Can you please help @anjneeksharma with this?

File: packages/feedback-service/src/resolver.ts [hunk 22]
```diff
@@ -24,29 +24,60 @@ export const FeedbackResolver = {
...
+            };
+            const graphql_api = process.env.GRAPHQL_API;
+            return fetch(graphql_api, requestOptions)
+              .then((result: any) => result.json())
+              .then((output: any) => {
+                const usersdata = output.data;
+                const createdby = usersdata['rhatUUID_' + response['createdBy']][0];
+                const updatedby = usersdata['rhatUUID_' + response['updatedBy']];
+                const res = { ...{ ...response }._doc, createdBy: createdby, updatedBy: updatedby };
+                return res;
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:68]:
```suggestion
                return { ...{ ...response }._doc, createdBy: createdby, updatedBy: updatedby };
```

Comment (deshmukhmayur) [reply packages/feedback-service/src/resolver.ts:68]: Same thing wherever applicable.

File: packages/feedback-service/package.json [hunk 23]
```diff
@@ -57,7 +57,8 @@
     "method-override": "3.0.0",
     "mongoose": "5.9.16",
     "morgan": "1.10.0",
-    "nodemailer": "6.4.8",
+    "node-fetch": "^2.6.0",
+    "nodemailer": "6.4.6",
```
Comment (deshmukhmayur) [inline packages/feedback-service/package.json:61]: Unused package. You can remove it.
Also make sure to run `npm install` after removing it so the package-lock.json is updated.

File: packages/feedback-service/src/resolver.ts [hunk 24]
```diff
@@ -1,76 +1,162 @@
...
+          const graphql = JSON.stringify({
+            query: formattedquery,
+            variables: {}
+          });
+          const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: graphql
+          };
+          const graphql_api = process.env.GRAPHQL_API;
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:27]: Should be using the environment variables for User Service.

```suggestion
          const graphql_api = `http://${process.env.USER_SERVICE_SERVICE_HOST}:${process.env.USER_SERVICE_SERVICE_PORT}/graphql`;
```

File: packages/feedback-service/src/resolver.ts [hunk 25]
```diff
@@ -1,76 +1,162 @@
...
+            const graphql = JSON.stringify({
+              query: formattedquery,
+              variables: {}
+            });
+            const requestOptions = {
+              method: 'POST',
+              headers: { 'Content-Type': 'application/json' },
+              body: graphql
+            };
+            const graphql_api = process.env.GRAPHQL_API;
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:60]:
```suggestion
            const graphql_api = `http://${process.env.USER_SERVICE_SERVICE_HOST}:${process.env.USER_SERVICE_SERVICE_PORT}/graphql`;
```

File: packages/feedback-service/src/resolver.ts [hunk 26]
```diff
@@ -1,76 +1,162 @@
...
           });
-      })
-      .catch((err: any) => {
-        throw err;
+          const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: graphql
+          };
+          const graphql_api = process.env.GRAPHQL_API;
```
Comment (deshmukhmayur) [inline packages/feedback-service/src/resolver.ts:91]:
```suggestion
          const graphql_api = `http://${process.env.USER_SERVICE_SERVICE_HOST}:${process.env.USER_SERVICE_SERVICE_PORT}/graphql`;
```

**Comment by riginoommen** [comment]
> @anjneeksharma The commit history within this PR looks quite polluted with multiple rebase commits. Can you please clean up the commit history for this PR? @riginoommen Can you please help @anjneeksharma with this?

Once you fix the existing issues you have to squah the commits into one meaningful commit 

Please see this - http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html .
If you face any issues regarding this we can check together.

**Comment by anjneeksharma** [comment]
@deshmukhmayur  i squashed all the commits, plz have another look.

**Comment by deshmukhmayur** [comment]
@anjneeksharma This branch has some conflicts. Can you rebase and update it?

**Comment by anjneeksharma** [comment]
@deshmukhmayur but its showing message no conflict.

**Comment by anjneeksharma** [comment]
@deshmukhmayur this is old PR so many versions in package.json are changed in other spa and service so getting many conflict while squashing commits.

**Comment by sayak-sarkar** [comment]
@anjneeksharma going by the commit history, it seems that the squashn was only partial. @riginoommen can you please work with @anjneeksharma today and get this resolved?

File: cli/generator-op-microservice/generators/microservice/templates/gql.service.ts [hunk 27]
```diff
@@ -71,15 +71,7 @@ const apollo = new ApolloServer({
...
-const server = process.env.NODE_ENV !== 'test'
-  ? https.createServer(
-    {
-      key: fs.readFileSync(process.env.SSL_KEY || ''),
-      cert: fs.readFileSync(process.env.SSL_CERT || '')
-    },
-    app
-  )
-  : http.createServer(app);
+const server =  http.createServer(app);
```
Comment (riginoommen) [inline cli/generator-op-microservice/generators/microservice/templates/gql.service.ts:74]: Why the feedback schema is dependend on the cli generator

File: cli/generator-op-microservice/generators/microservice/templates/.env.example [hunk 28]
```diff
@@ -14,10 +14,6 @@ DB_NAME=db_name
 # DB_USER=
 # DB_PASSWORD=
```
Comment (riginoommen) [inline cli/generator-op-microservice/generators/microservice/templates/.env.example:16]: You need to revert the changes on this file

File: .gitignore [hunk 29]
```diff
@@ -60,3 +60,6 @@ typings/
 
 # next.js build output
 .next
+
+# Temporary files
+tmp/
```
Comment (riginoommen) [inline .gitignore:65]: i think this tmp folder is not dependent on the feedback

File: cli/generator-op-microservice/generators/microservice/templates/rest.service.ts [hunk 30]
```diff
@@ -13,11 +13,6 @@ import mongoose from 'mongoose';
 
 import <%= apiName %>Router from './src/api';
```
Comment (riginoommen) [inline cli/generator-op-microservice/generators/microservice/templates/rest.service.ts:15]: Please checkout this files and revert the changes

File: cli/generator-op-microservice/package-lock.json [hunk 31]
```diff
@@ -434,107 +434,56 @@
       "dev": true
     },
     "@jest/console": {
-      "version": "25.5.0",
-      "resolved": "https://registry.npmjs.org/@jest/console/-/console-25.5.0.tgz",
-      "integrity": "sha512-T48kZa6MK1Y6k4b89sexwmSF4YLeZS/Udqg3Jj3jG/cHH+N/sLFCEoXEDMOKugJQ9FxPN1osxIknvKkxt6MKyw==",
+      "version": "26.0.1",
```
Comment (riginoommen) [inline cli/generator-op-microservice/package-lock.json:437]: Please checkout this files and revert the changes

File: cli/generator-op-microservice/package.json [hunk 32]
```diff
@@ -24,14 +24,14 @@
   "devDependencies": {
```
Comment (riginoommen) [inline cli/generator-op-microservice/package.json:24]: Please checkout this files and revert the changes

File: docker-compose.yml [hunk 33]
```diff
@@ -4,11 +4,13 @@ services:
...
+    environment:
+      SSI_HOSTNAME: /.include/nav
     volumes:
-      - ./packages/ssi-service/dist:/var/www/html/templates
-      - ./packages/ssi-service/public/index.html:/usr/share/httpd/noindex/index.html
+      - ./packages/ssi-service/dist:/var/www/html/.include/nav
+      - ./packages/ssi-service/public/index.html:/var/www/html/index.html
     ports:
-      - 5000:8080
+      - 5500:8080
```
Comment (riginoommen) [inline docker-compose.yml:14]: Please checkout this files and revert the changes

File: package.json [hunk 34]
```diff
@@ -40,14 +40,14 @@
...
+    "@typescript-eslint/eslint-plugin": "3.0.2",
+    "@typescript-eslint/parser": "3.0.2",
+    "eslint": "7.1.0",
     "eslint-config-standard": "14.1.1",
     "eslint-plugin-import": "2.20.2",
     "eslint-plugin-node": "11.1.0",
     "eslint-plugin-promise": "4.2.1",
     "eslint-plugin-standard": "4.0.1",
-    "lerna": "3.20.2"
+    "lerna": "3.22.0"
```
Comment (riginoommen) [inline package.json:51]: Please checkout this files and revert the changes

File: packages/api-gateway-service/.env.example [hunk 35]
```diff
@@ -8,11 +8,6 @@ PRODUCTION=false
 API=https://localhost:4000
 GRAPHQL_API=https://localhost:4000/graphql
 
-## Certificate Configuration
```
Comment (riginoommen) [inline packages/api-gateway-service/.env.example:11]: Please checkout this files and revert the changes

File: package.json (hunk 34 above)
Comment (anjneeksharma) [reply package.json:51]: @this is older PR plz review new one 
https://github.com/1-Platform/one-platform/pull/407

File: packages/api-gateway-service/.env.example [hunk 36]
```diff
@@ -8,11 +8,6 @@ PRODUCTION=false
 API=https://localhost:4000
 GRAPHQL_API=https://localhost:4000/graphql
```
Comment (riginoommen) [inline packages/api-gateway-service/.env.example:9]: Please checkout this files and revert the changes

File: packages/api-gateway-service/.env.example [hunk 37]
```diff
@@ -21,4 +16,7 @@ COOKIE_PARSER=
 
 ## Dependent Microservices
 ## User Group
-# USER_GROUP="https://localhost:8080/graphql"
+# USER_SERVICE_SERVICE_HOST="localhost:8080"
```
Comment (riginoommen) [inline packages/api-gateway-service/.env.example:19]: Please checkout this files and revert the changes

File: packages/api-gateway-service/package-lock.json [hunk 38]
```diff
@@ -5,44 +5,46 @@
   "requires": true,
   "dependencies": {
     "@apollo/federation": {
-      "version": "0.14.1",
-      "resolved": "https://registry.npmjs.org/@apollo/federation/-/federation-0.14.1.tgz",
-      "integrity": "sha512-QuX2O3xO6mcTZdqttxHaMKWgq1v0nYRiDLe4k7DwAxVtb9nF8lsJDlup4Zicx3LBYhBCGQvumrYILuF/Amn6WQ==",
+      "version": "0.16.2",
```
Comment (riginoommen) [inline packages/api-gateway-service/package-lock.json:8]: Please checkout this files and revert the changes

File: packages/api-gateway-service/package.json [hunk 39]
```diff
@@ -35,51 +35,51 @@
   "contributors": [],
   "license": "MIT",
   "dependencies": {
-    "@apollo/gateway": "0.14.1",
-    "@types/jsonwebtoken": "8.3.9",
+    "@apollo/gateway": "0.16.2",
```
Comment (riginoommen) [inline packages/api-gateway-service/package.json:38]: Please checkout this files and revert the changes

File: packages/api-gateway-service/service.ts [hunk 40]
```diff
@@ -77,25 +69,27 @@ const server = process.env.NODE_ENV !== 'test'
...
+            return res.status( 403 ).json( err );
+          } else if ( err && err.name === 'JsonWebTokenError' ) {
+            return res.status( 403 ).json( err );
+          } else if ( !err ) {
+            decodedPayload = JSON.stringify( payload );
+            next();
+          }
+        } );
+      } catch ( err ) {
+        return res.status( 403 ).json( err );
```
Comment (riginoommen) [inline packages/api-gateway-service/service.ts:90]: Please checkout this files and revert the changes

File: packages/api-gateway-service/src/helpers.ts [hunk 41]
```diff
@@ -2,7 +2,10 @@ const fetch = require( 'node-fetch' );
 
 /* This consists of the list of microservices which needs to be part of the gateway */
 export const serviceList = [
-  { name: 'User Group', url: process.env.USER_GROUP }
+  { name: 'User Service', url: `http://${ process.env.USER_SERVICE_SERVICE_HOST }:8080/graphql` },
+  { name: 'Home Service', url: `http://${ process.env.HOME_SERVICE_SERVICE_HOST }:8080/graphql` },
+  { name: 'Feedback Service', url: `http://${ process.env.FEEDBACK_SERVICE_SERVICE_HOST }:8080/graphql` },
+  { name: 'Notification Service', url: `http://${ process.env.NOTIFICATIONS_SERVICE_SERVICE_HOST }:8080/graphql` },
```
Comment (riginoommen) [inline packages/api-gateway-service/src/helpers.ts:8]: Please checkout this files and revert the changes because this are all changed

File: packages/feedback-service/package.json [hunk 42]
```diff
@@ -35,56 +35,56 @@
   "contributors": [],
   "license": "MIT",
   "dependencies": {
-    "@apollo/federation": "0.14.1",
+    "@apollo/federation": "0.16.2",
```
Comment (riginoommen) [inline packages/feedback-service/package.json:38]: I have removed this package from the code

File: packages/feedback-service/service.ts [hunk 43]
```diff
@@ -3,8 +3,8 @@ import { ApolloServer } from 'apollo-server-express';
 import fs from 'fs';
 import https from 'https';
 import http from 'http';
-const { ApolloLogExtension } = require('apollo-log');
-const { buildFederatedSchema } = require("@apollo/federation");
+const { ApolloLogExtension } = require( 'apollo-log' );
+import { buildFederatedSchema } from '@apollo/federation';
```
Comment (riginoommen) [inline packages/feedback-service/service.ts:7]: Federation has been removed

File: packages/feedback-service/service.ts [hunk 44]
```diff
@@ -14,64 +14,55 @@ import { FeedbackResolver as resolver } from './src/resolver';
...
 /* Defining the Apollo Server */
 const apollo = new ApolloServer( {
   playground: process.env.NODE_ENV !== 'production',
   schema: buildFederatedSchema( [ {
     typeDefs: gqlSchema,
     resolvers: resolver
-  }]),
-  
-  formatError: error => ({
+  } ] ),
```
Comment (riginoommen) [inline packages/feedback-service/service.ts:45]: federated schema has been changed to simple graphql server

**Comment by anjneeksharma** [comment]
created another PR for same purpose.
